lib/winrm/transport/file_transporter.rb

```diff
@@ -253,7 +252,10 @@ def decode_files(files)
           vars = %{$hash_file = "#{hash_file}"\n}
 
           output = service.run_powershell_script(
-            [vars, decode_files_script].join("\n")
+            [vars,
+             ". #{decode_files_script}",
+             "Decode-Files (Invoke-Input $hash_file) | ConvertTo-Csv -NoTypeInformation"
```

Comment:
Yeah, when we ran the Decode-Files line in the script it makes testing a bit tougher.